usr/src/cmd/iscsid/iscsi-initiator

```diff
@@ -170,6 +171,23 @@ mount_iscsi() {
 }
 
 umount_iscsi () {
+	# In case any non-"zones" zpools reside on iscsi drives, export
+	# all of them now so that the system won't hang later in the shutdown
+	# sequence when those zpools need to be unmounted.
+	zpools=$(zpool list -Ho name)
+	if [ -n $zpools ]; then
+		for pool in $zpools; do
+			if [ "$pool" == "zones" ]; then
+				continue
+			fi
+			zpool export $pool
+			rc=$?
+			if [ $rc -ne 0 ]; then
+				return $rc
```

Comment:
It might be nice to have a log line here. Would this get scooped up by the iscsi-initiator SMF service log?

Do you think we should still continue the loop and try to export any other pools we find rather than stopping on the first error?